src/App.tsx

```diff
@@ -1,19 +1,191 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
+import { TodosList } from './components/TodosList/TodosList';
 
-import users from './api/users';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/todo';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  users: User[];
+  todos: Todo[];
+  todoTitle: string;
+  todoUserId: number;
+  todoUserError: string,
+  todoTitleError: string,
+  maxLengthInput: number,
 };
 
+const preparedTodos = todosFromServer.map((todo) => {
+  const user = usersFromServer.find(({ id }) => todo.userId === id) || null;
+
+  return { ...todo, user };
+});
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    users: usersFromServer,
+    todos: preparedTodos,
+    todoTitle: '',
+    todoUserId: 0,
+    todoUserError: '',
+    todoTitleError: '',
+    maxLengthInput: 15,
+  };
+
+  getMaxId = () => {
+    const ids = this.state.todos.map(({ id }) => id) || 1;
+
+    return Math.max(...ids);
+  };
+
+  resetForm = () => {
+    this.setState({
+      todoTitle: '',
+      todoUserId: 0,
+    });
+  };
+
+  handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const {
+      users,
+      todoUserId,
+      todoTitle,
+      maxLengthInput,
+    } = this.state;
+
+    if (todoUserId === 0) {
+      this.setState({ todoUserError: 'Please choose a user' });
+
+      return;
+    }
+
+    if (todoTitle.length === 0) {
+      this.setState({ todoTitleError: 'Please enter the title' });
+
+      return;
+    }
+
+    if (todoTitle.length > maxLengthInput) {
+      this.setState({ todoTitleError: `Maximum number of characters ${maxLengthInput}. You - ${todoTitle.length}` });
+
+      return;
+    }
+
+    if (!todoTitle.match(/^[аА-яЯaA-zZ0-9]/)) {
+      this.setState({
+        todoTitleError: 'Invalid characters: letters (ru and en), digits and spaces are available',
+      });
+
+      return;
+    }
```

Comment:
it seems like this could be taken out into a separate function. this logic is the validation